apps/test/unit/templates/teacherDashboard/TeacherDashboardHeaderTest.js

```diff
@@ -5,29 +5,39 @@ import {expect} from '../../../util/deprecatedChai';
 import {UnconnectedTeacherDashboardHeader as TeacherDashboardHeader} from '@cdo/apps/templates/teacherDashboard/TeacherDashboardHeader';
 import FontAwesome from '@cdo/apps/templates/FontAwesome';
 
-const MOCK_SECTIONS = {
-  1: {
-    id: 1,
-    name: 'intro to computer science I'
+// The array only needs to conform to sectionForDropdownShape,
+// but for convenience we're using the first entry as the selectedSection,
+// which needs to conform to the whole sectionShape
+const MOCK_SECTIONS = [
+  {
+    id: 3,
+    name: 'intro to computer science III',
+    isAssigned: false,
+    stageExtras: true,
+    pairingAllowed: true,
+    studentCount: 5,
+    code: 'VQGSJR',
+    providerManaged: false
   },
-  2: {
+  {
     id: 2,
-    name: 'intro to computer science II'
+    name: 'intro to computer science II',
+    isAssigned: false
   },
-  3: {
-    id: 3,
-    name: 'hidden section',
-    hidden: true
```

Comment:
Making this section not be hidden affects the test on line 67 (after your changes) - "renders dropdown button with links to sections, highlighting current section, ignoring hidden section". If none of the mock sections are hidden, we aren't actually testing whether it properly ignores hidden sections.

Reply:
hmm. as i mentioned in my comment to Max, throughout the rest of our code, filtering out hidden sections is done somewhere in the "data layer" (generally on the backend), so all our section dropdowns assume they are passed exactly what they need (though not necessarily pre-sorted). to stick with that convention of an unopinionated view layer in regards to filtering, i moved the filtering into the `connect` method with a comment about this component being a special case re. needing to filter its data. so basically the `TeacherDashboardHeader` displays whatever it's given _by design/convention_. hence, the test was updated to have its mock data already listified, filtered, and sorted.

a few options here:

1) i can move the filtering back into the component logic. this dropdown is already an exception in that it requires a redux tree with unfiltered section data, and that "filter in data layer" convention isn't exactly formalized anywhere anyway.

2) i can leverage some of the techniques from `teacherSectionsReduxTest` to test the component with an actual redux store rather than just testing the unconnected component.

3) i can add a comment in the test about the component expecting pre-filtered/sorted data and update the description of the test that was testing the filtering, and leave the data as-is.

this issue highlights the fact that we don't seem to have a clear protocol around testing unconnected vs. connected components. we often put quite a bit of business logic in the `connect` method, but (in this case at least) we are only testing the unconnected version.

as long as we are considering component tests to essentially be rendering tests with explicitly defined test data, i think option #3 would be the most consistent. but really we should probably be moving toward making sure we always test the connected components so we don't miss any bugs/regressions in `connect'. and with that in mind, option #2 would probably be the right way to go. option #1 would be a nice easy way to kick this conversation down the road!

Reply:
lmk what you think @clareconstantine!

Reply:
3 sounds good to me. Thanks for thinking through all the options!

Reply:
👍 thanks for pointing this out! i had totally missed the implications of removing that hidden section from the mock data.